--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -21 +21,4 @@
-mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "/system");
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+ui_print("Installing ROM as secondary");
+run_program("/tmp/dualboot.sh", "mount-system", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/system");
@@ -23,2 +26,2 @@
-unmount("/system");
-format("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "0", "/system");
+run_program("/tmp/dualboot.sh", "unmount-system");
+ui_print("Formatting /system/dual for dual boot");
@@ -25,0 +29,3 @@
+delete_recursive("/system/dual");
+unmount("/system");
+run_program("/tmp/dualboot.sh", "mount-system", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/system");
@@ -162 +168,2 @@
-unmount("/system");
+run_program("/tmp/dualboot.sh", "set-secondary-kernel");
+run_program("/tmp/dualboot.sh", "unmount-system");
