--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -1,11 +1,15 @@
-mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "/system");
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+ui_print("Installing ROM as secondary");
+run_program("/tmp/dualboot.sh", "mount-system", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/system");
+run_program("/tmp/dualboot.sh", "mount-data", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/userdata");
 package_extract_dir("system/addon.d", "/system/addon.d");
 package_extract_file("system/bin/backuptool.sh", "/tmp/backuptool.sh");
 package_extract_file("system/bin/backuptool.functions", "/tmp/backuptool.functions");
 set_perm(0, 0, 0777, "/tmp/backuptool.sh");
 set_perm(0, 0, 0644, "/tmp/backuptool.functions");
 run_program("/tmp/backuptool.sh", "backup");
-unmount("/system");
+run_program("/tmp/dualboot.sh", "unmount-system");
 show_progress(0.500000, 0);
 ui_print("");
 ui_print("");
@@ -15,9 +19,11 @@
 ui_print("");
 ui_print("");
 ui_print("Format and Mount System, Mount Data...");
-format("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "0", "/system");
+ui_print("Formatting /system/dual for dual boot");
 mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "/system");
-mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/userdata", "/data");
+delete_recursive("/system/dual");
+unmount("/system");
+run_program("/tmp/dualboot.sh", "mount-system", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/system");
 package_extract_dir("system", "/system");
 package_extract_dir("data", "/data");
 ui_print("Symlinks and permissions...");
@@ -161,6 +167,7 @@
 show_progress(0.200000, 10);
 ui_print("Flashing Boot Image...");
 package_extract_file("boot.img", "/dev/block/platform/msm_sdcc.1/by-name/boot");
+run_program("/tmp/dualboot.sh", "set-secondary-kernel");
 show_progress(0.100000, 0);
-unmount("/data");
-unmount("/system");
+run_program("/tmp/dualboot.sh", "unmount-data");
+run_program("/tmp/dualboot.sh", "unmount-system");
