--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -1,9 +1,14 @@
 ui_print("***********************************************");
 ui_print("        Google Apps for Android 4.3.0");
 ui_print("***********************************************");
+ui_print("       Installing for SECONDARY OS/ROM");
+ui_print("***********************************************");
+
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+run_program("/tmp/dualboot.sh", "mount-system", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/system");
 
 ui_print("Installing files...");
-run_program("/sbin/busybox", "mount", "/system");
 show_progress(1, 15);
 delete("/system/app/Provision.apk","/system/app/QuickSearchBox.apk","/system/app/SetupWizard.apk","/system/app/Velvet.apk","/system/app/Vending.apk","/system/app/BrowserProviderProxy.apk","/system/app/PartnerBookmarksProvider.apk","/system/app/Gmail2.apk");
 package_extract_dir("system", "/system");
@@ -15,5 +20,7 @@
 show_progress(1, 15);
 ui_print("Cleaning up...");
 set_perm_recursive(0, 0, 0755, 0644, "/system/app");
-run_program("/sbin/busybox", "umount", "/system");
+
+run_program("/tmp/dualboot.sh", "unmount-system");
+
 ui_print("Installation complete!");
