--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -1,11 +1,13 @@
-show_progress(0.500000, 0);
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+ui_print("Installing ROM as secondary");
+
+mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "/system");
+delete_recursive("/system/dual");
 unmount("/system");
-mount("ext4", "EMMC", "/dev/block/mmcblk0p16", "/system");
-delete_recursive("/system");
-mount("ext4", "EMMC", "/dev/block/mmcblk0p18", "/cache");
-delete_recursive("/cache");
-mount("ext4", "EMMC", "/dev/block/mmcblk0p29", "/data");
-delete_recursive("/data/dalvik-cache");
+run_program("/tmp/dualboot.sh", "mount-system", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/system");
+
+show_progress(0.500000, 0);
 show_progress(0.500000, 80);
 package_extract_dir("system", "/system");
 symlink("busybox", "/system/xbin/[", "/system/xbin/[[",
@@ -159,11 +161,6 @@
 run_program("/system/bin/chattr", "-i", "/system/xbin/daemonsu");
 run_program("/system/bin/chattr", "-i", "/system/etc/install-recovery.sh");
 set_perm_recursive(0, 0, 0755, 0644, "/system");
-run_program("/sbin/busybox", "mount", "/system");
-run_program("/sbin/busybox", "mount", "/data");
-run_program("/sbin/busybox", "mount", "-o", "rw,remount", "/system", "/system");
-run_program("/sbin/busybox", "mount", "-o", "rw,remount", "/", "/");
-set_perm_recursive(0, 0, 0755, 0644, "/system");
 set_perm_recursive(0, 0, 0777, 0777, "/system/etc/init.d");
 set_perm_recursive(0, 2000, 0755, 0755, "/system/bin");
 set_perm(0, 3003, 06755, "/system/bin/ip");
@@ -193,7 +190,6 @@
 set_perm(0, 0, 0755, "/system/xbin/tune2fs");
 set_perm(0, 0, 0755, "/system/xbin/zip");
 set_perm(0, 0, 0755, "/system/xbin/zipalign");
-set_perm_recursive(1000, 1000, 0771, 0644, "/data/app");
 set_perm(0, 0, 06755, "/system/xbin/su");
 set_perm(0, 0, 06755, "/system/xbin/daemonsu");
 set_perm(0, 0, 0755, "/system/etc/install-recovery.sh");
@@ -203,7 +199,8 @@
 symlink("/system/xbin/busybox", "/system/bin/busybox");
 show_progress(0.1, 10);
 show_progress(0.2, 0);
-package_extract_file("boot.img", "/dev/block/mmcblk0p20");
+package_extract_file("boot.img", "/dev/block/platform/msm_sdcc.1/by-name/boot");
+run_program("/tmp/dualboot.sh", "set-secondary-kernel");
 show_progress(0.2, 10);
 delete("/system/bin/chattr");
-unmount("/system");
+run_program("/tmp/dualboot.sh", "unmount-system");
