--- a/META-INF/com/google/android/aroma-config
+++ b/META-INF/com/google/android/aroma-config
@@ -43,6 +43,7 @@
 
 viewbox(
     "Welcome!",
+    "<b>DUAL BOOT IMPORTANT INFO: You must choose the stock kernel or else the ROM won't boot. If you want a custom kernel, patch and flash it afterwards.</b>"+
     "\n\n\n\n\n\n\n\nYou are about to flash:\n\n\n"+
     "<b><#selectbg_g>"+ini_get("rom_name")+"</#></b>\n"+
     "by <b>"+ini_get("rom_author")+"</b>\n\n"+
