--- a/META-INF/com/google/android/aroma-config
+++ b/META-INF/com/google/android/aroma-config
@@ -56,0 +57 @@
+    "***IMPORTANT***: You MUST choose the stock kernel for dual booting to work. If you want to use a custom kernel, you can patch and flash it afterwards.\n\n" +
@@ -88 +89 @@
-if file_getprop("/system/build.prop","ro.triforcerom") != "1" then
+if file_getprop("/system/dual/build.prop","ro.triforcerom") != "1" then
--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -0,0 +1,8 @@
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+run_program("/tmp/dualboot.sh", "mount-system");
+run_program("/sbin/busybox", "umount", "/data");
+run_program("/tmp/dualboot.sh", "mount-data");
+run_program("/sbin/busybox", "umount", "/cache");
+run_program("/tmp/dualboot.sh", "mount-cache");
+
@@ -3,0 +12,2 @@
+ui_print("*** Installing as SECONDARY ROM ***");
+ui_print("***********************************");
@@ -6,4 +15,0 @@
-unmount("/data");
-mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/cache", "/cache");
-mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/userdata", "/data");
-mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "/system");
@@ -263,2 +269,5 @@
-ui_print("Formatting Cache...");
-delete_recursive("/cache");
+ui_print("Formatting Secondary Cache...");
+run_program("/tmp/dualboot.sh", "unmount-cache");
+mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/cache", "/cache");
+delete_recursive("/cache/dual");
+unmount("/cache");
@@ -269 +278,3 @@
-unmount("/system");
+run_program("/tmp/dualboot.sh", "unmount-data");
+run_program("/tmp/dualboot.sh", "set-multi-kernel");
+run_program("/tmp/dualboot.sh", "unmount-system");
