From cc038e262dde2dfea4cb102f67d5ac7067c06d0f Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <chenxiaolong@cxl.epac.to>
Date: Thu, 2 Feb 2017 20:00:31 -0500
Subject: [PATCH 3/3] Android does not define __STDC_VERSION__

---
 libexfat/compiler.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libexfat/compiler.h b/libexfat/compiler.h
index e4f40de..299d888 100644
--- a/libexfat/compiler.h
+++ b/libexfat/compiler.h
@@ -24,7 +24,7 @@
 #ifndef COMPILER_H_INCLUDED
 #define COMPILER_H_INCLUDED
 
-#if __STDC_VERSION__ < 199901L
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ < 199901L
 #error C99-compliant compiler is required
 #endif
 
-- 
2.14.3

