From 00760cd7db62abc2335fe1c074300f4bb4a82e35 Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <andrewgunnerson@gmail.com>
Date: Sat, 15 Nov 2014 14:39:40 -0500
Subject: [PATCH] Change statfs.f_flag -> statfs.f_flags

---
 libarchive/archive_read_disk_posix.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/libarchive/archive_read_disk_posix.c b/libarchive/archive_read_disk_posix.c
index 059219b..cb52272 100644
--- a/libarchive/archive_read_disk_posix.c
+++ b/libarchive/archive_read_disk_posix.c
@@ -1628,13 +1628,13 @@ setup_current_filesystem(struct archive_read_disk *a)
 		t->current_filesystem->incr_xfer_size = sfs.f_bsize;
 #endif
 	}
-	if (sfs.f_flag & ST_LOCAL)
+	if (sfs.f_flags & ST_LOCAL)
 		t->current_filesystem->remote = 0;
 	else
 		t->current_filesystem->remote = 1;
 
 #if defined(ST_NOATIME)
-	if (sfs.f_flag & ST_NOATIME)
+	if (sfs.f_flags & ST_NOATIME)
 		t->current_filesystem->noatime = 1;
 	else
 #endif
@@ -1750,7 +1750,7 @@ setup_current_filesystem(struct archive_read_disk *a)
 	}
 
 #if defined(ST_NOATIME)
-	if (svfs.f_flag & ST_NOATIME)
+	if (svfs.f_flags & ST_NOATIME)
 		t->current_filesystem->noatime = 1;
 	else
 #endif
@@ -1834,7 +1834,7 @@ setup_current_filesystem(struct archive_read_disk *a)
 	}
 
 #if defined(ST_NOATIME)
-	if (sfs.f_flag & ST_NOATIME)
+	if (sfs.f_flags & ST_NOATIME)
 		t->current_filesystem->noatime = 1;
 	else
 #endif
-- 
2.1.3

