--- a/META-INF/com/google/android/update-binary
+++ b/META-INF/com/google/android/update-binary
@@ -140,10 +140,8 @@
 ui_print "*********************"
 
 ui_print "- Mounting /system, /data and rootfs"
-mount /system
-mount /data
-mount -o rw,remount /system
-mount -o rw,remount /system /system
+/update-binary-tool mount /system
+/update-binary-tool mount /data
 mount -o rw,remount /
 mount -o rw,remount / /
 
@@ -397,8 +395,8 @@
 LD_LIBRARY_PATH=$SYSTEMLIB /system/xbin/su --install
 
 ui_print "- Unmounting /system and /data"
-umount /system
-umount /data
+/update-binary-tool unmount /system
+/update-binary-tool unmount /data
 
 ui_print "- Done !"
 exit 0
