--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -1,16 +1,22 @@
-ui_print("mounting system");
-run_program("/sbin/busybox", "mount", "/system");
+mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "/system");
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+unmount("/system");
+ui_print("Installing kernel for secondary ROM");
+run_program("/tmp/dualboot.sh", "mount-system");
+
 ui_print("copying modules");
 ui_print("");
 package_extract_dir("system", "/system");
 
 ui_print("unmounting system");
 ui_print("");
-unmount("/system");
 
 ui_print("Flashing Kernel");
 ui_print("");
 
 package_extract_file("boot.lok", "/dev/block/platform/msm_sdcc.1/by-name/boot");
+run_program("/tmp/dualboot.sh", "set-multi-kernel");
+run_program("/tmp/dualboot.sh", "unmount-system");
 show_progress(0.100000, 0);
 
