From 1575a66295498225c4489cf73ca45d003720960c Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <chenxiaolong@cxl.epac.to>
Date: Sun, 26 Mar 2017 15:53:30 -0400
Subject: [PATCH] Make this NDK buildable

Signed-off-by: Andrew Gunnerson <chenxiaolong@cxl.epac.to>
---
 Android.mk | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/Android.mk b/Android.mk
index 76df4f1..0c27755 100644
--- a/Android.mk
+++ b/Android.mk
@@ -47,23 +47,3 @@ LOCAL_CFLAGS := \
 	$(common_CFLAGS)
 
 include $(BUILD_STATIC_LIBRARY)
-
-# Dynamic library for the device
-include $(CLEAR_VARS)
-
-LOCAL_MODULE := libdrm
-
-LOCAL_SRC_FILES := $(filter-out %.h,$(LIBDRM_FILES))
-LOCAL_EXPORT_C_INCLUDE_DIRS := \
-	$(LOCAL_PATH) \
-	$(LOCAL_PATH)/include/drm
-
-LOCAL_C_INCLUDES := \
-	$(LOCAL_PATH)/include/drm
-
-LOCAL_CFLAGS := \
-	$(common_CFLAGS)
-
-include $(BUILD_SHARED_LIBRARY)
-
-include $(call all-makefiles-under,$(LOCAL_PATH))
-- 
2.9.3

