From 730c474a165784faaa47d453fc389219eb3f3e4d Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <andrewgunnerson@gmail.com>
Date: Sat, 15 Nov 2014 16:16:18 -0500
Subject: [PATCH 2/2] libxml2: Remove BUILD_HOST_STATIC_LIBRARY

---
 Android.mk | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/Android.mk b/Android.mk
index 883f898..dfc3f16 100644
--- a/Android.mk
+++ b/Android.mk
@@ -90,10 +90,10 @@ include $(BUILD_STATIC_LIBRARY)
 # For the host
 # ========================================================
 
-include $(CLEAR_VARS)
-LOCAL_SRC_FILES := $(common_SRC_FILES)
-LOCAL_C_INCLUDES += $(common_C_INCLUDES)
-LOCAL_CFLAGS += $(DISABLED_WARNING_FLAGS)
-LOCAL_SHARED_LIBRARIES += $(common_SHARED_LIBRARIES)
-LOCAL_MODULE:= libxml2
-include $(BUILD_HOST_STATIC_LIBRARY)
+#include $(CLEAR_VARS)
+#LOCAL_SRC_FILES := $(common_SRC_FILES)
+#LOCAL_C_INCLUDES += $(common_C_INCLUDES)
+#LOCAL_CFLAGS += $(DISABLED_WARNING_FLAGS)
+#LOCAL_SHARED_LIBRARIES += $(common_SHARED_LIBRARIES)
+#LOCAL_MODULE:= libxml2
+#include $(BUILD_HOST_STATIC_LIBRARY)
-- 
2.1.3

