--- a/Lib/plat-linux/DLFCN.py
+++ b/Lib/plat-linux/DLFCN.py
@@ -75,9 +75,20 @@
 
 # Included from bits/dlfcn.h
 RTLD_LAZY = 0x00001
+#ifdef __ANDROID__
+RTLD_NOW = 0x00000
+RTLD_BINDING_MASK = 0x0
+RTLD_NOLOAD = 0x00000
+RTLD_GLOBAL = 0x00002
+#else
 RTLD_NOW = 0x00002
 RTLD_BINDING_MASK = 0x3
 RTLD_NOLOAD = 0x00004
 RTLD_GLOBAL = 0x00100
+#endif
 RTLD_LOCAL = 0
+#ifdef __ANDROID__
+RTLD_NODELETE = 0x00000
+#else
 RTLD_NODELETE = 0x01000
+#endif
