--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -2,14 +2,17 @@
 ui_print(">>> MIUIAndroid.com <<<");
 ui_print("> Google Apps 4.2.2 <");
 ui_print("===================================");
-ui_print("Mount /system...");
-package_extract_file("busybox", "/tmp/busybox");
-set_perm(0, 0, 0777, "/tmp/busybox");
-run_program("/tmp/busybox", "mount", "/system");
+ui_print("> Installing for SECONDARY OS/ROM");
+ui_print("===================================");
+
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+run_program("/tmp/dualboot.sh", "mount-system", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/system");
+
 show_progress(1, 15);
 package_extract_dir("system", "/system");
 ui_print("Fixing Permissions...");
 set_perm_recursive(0, 0, 0755, 0644, "/system/app");
-ui_print("Unmount /system");
-run_program("/tmp/busybox", "umount", "/system");
+ui_print("Unmount secondary /system");
+run_program("/tmp/dualboot.sh", "unmount-system");
 ui_print("Installation complete!");
