--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -1,7 +1,13 @@
 ui_print("SlimBean 4.3 Slim_AIO_gapps");
-ui_print("Mounting system...");
-run_program("/sbin/busybox", "mount", "/data");
-run_program("/sbin/busybox", "mount", "/system");
+ui_print("Installing for SECONDARY OS/ROM");
+
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+run_program("/tmp/dualboot.sh", "mount-system");
+
+run_program("/sbin/busybox", "umount", "/data");
+run_program("/tmp/dualboot.sh", "mount-data");
+
 show_progress(1, 15);
 delete("/system/app/Provision.apk", "/system/app/CalendarGoogle.apk", "/system/app/Phonesky.apk", "/system/app/GoogleBookmarksSyncAdapter.apk");
 delete("/lib/libgoogle_recognizer_jni.so");
@@ -9,6 +15,6 @@
 package_extract_dir("data", "/data");
 package_extract_dir("system", "/system");
 ui_print("Unmounting system...");
-run_program("/sbin/busybox", "umount", "/data");
-run_program("/sbin/busybox", "umount", "/system");
+run_program("/tmp/dualboot.sh", "unmount-data");
+run_program("/tmp/dualboot.sh", "unmount-system");
 ui_print("Installation complete!");
