--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -1,16 +1,17 @@
+ui_print("Installing for SECONDARY OS/ROM");
+
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+run_program("/tmp/dualboot.sh", "mount-system", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/system");
+run_program("/sbin/busybox", "umount", "/data");
+run_program("/tmp/dualboot.sh", "mount-data", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/userdata");
+run_program("/sbin/busybox", "umount", "/cache");
+run_program("/tmp/dualboot.sh", "mount-cache", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/cache");
+
 # Start
 set_progress(0.05);
 ui_print("Installing SlimBean aroma selectable gapps");
 
-# Mount
-ui_print("Setting up required tools...");
-package_extract_file("utils/mount", "/tmp/mount");
-set_perm(0, 0, 0755, "/tmp/mount");
-ui_print("Mounting Partitions");
-run_program("/tmp/mount", "/system");
-run_program("/tmp/mount", "/data");
-run_program("/tmp/mount", "/cache");
-
 # Install System Apps
 set_progress(0.03);
 ui_print("Installing Core Apps [Required]");
@@ -149,7 +150,9 @@
 delete_recursive("/tmp/aroma");
 
 # Unmount
-unmount("/data");
+run_program("/tmp/dualboot.sh", "unmount-data");
+run_program("/tmp/dualboot.sh", "unmount-cache");
+run_program("/tmp/dualboot.sh", "unmount-system");
 
 # Finish
 ui_print("Finished Install");
