--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -1,5 +1,9 @@
-ui_print("Mounting system...");
-run_program("/sbin/busybox", "mount", "/system");
+ui_print("Installing for SECONDARY OS/ROM");
+
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+run_program("/tmp/dualboot.sh", "mount-system");
+
 show_progress(1, 15);
 delete("/system/app/Provision.apk","/system/app/QuickSearchBox.apk","/system/app/SetupWizard.apk","/system/app/Velvet.apk","/system/app/Vending.apk","/system/app/BrowserProviderProxy.apk","/system/app/PartnerBookmarksProvider.apk");
 ui_print("Copying files...");
@@ -9,5 +13,5 @@
 ui_print("Fixing Permissions...");
 set_perm_recursive(0, 0, 0755, 0644, "/system/app");
 ui_print("Unmounting system...");
-run_program("/sbin/busybox", "umount", "/system");
+run_program("/tmp/dualboot.sh", "unmount-system");
 ui_print("Installation complete!");
