--- a/META-INF/com/google/android/update-binary
+++ b/META-INF/com/google/android/update-binary
@@ -23,10 +23,7 @@
 ui_print "SuperSU installer ZIP"
 ui_print "*********************"
 
-ui_print "- Mounting /system, /data and rootfs"
-mount /system
-mount /data
-mount -o rw,remount /system /system
+ui_print "- Mounting rootfs"
 mount -o rw,remount / /
 
 ABI=$(cat /default.prop | grep ro.product.cpu.abi= | dd bs=1 skip=19 count=3)
@@ -42,6 +39,11 @@
 cd supersu
 unzip -o "$ZIP"
 
+ui_print "- Installing for secondary ROM"
+set_perm 0 0 0777 /tmp/supersu/dualboot.sh
+/tmp/supersu/dualboot.sh mount-system ext4 /dev/block/platform/msm_sdcc.1/by-name/system
+/tmp/supersu/dualboot.sh mount-data ext4 /dev/block/platform/msm_sdcc.1/by-name/userdata
+
 BIN=/tmp/supersu/$ARCH
 COM=/tmp/supersu/common
 
@@ -140,8 +142,8 @@
 ch_con /system/app/YouTube.apk
 
 ui_print "- Unmounting /system and /data"
-umount /system
-umount /data
+/tmp/supersu/dualboot.sh unmount-data
+/tmp/supersu/dualboot.sh unmount-system
 
 ui_print "- Done !"
 exit 0
