--- a/init.target.rc
+++ b/init.target.rc
@@ -39,14 +39,15 @@
 
 on fs
     mount_all fstab.qcom
+    exec /sbin/busybox-static sh /init.dualboot.mounting.sh
     setprop ro.crypto.fuse_sdcard true
 
     # Keeping following partitions outside fstab file. As user may not have
     # these partition flashed on the device. Failure to mount any partition in fstab file
     # results in failure to launch late-start class.
-    wait /dev/block/platform/msm_sdcc.1/by-name/cache
-    check_fs /dev/block/platform/msm_sdcc.1/by-name/cache ext4
-    mount ext4 /dev/block/platform/msm_sdcc.1/by-name/cache /cache nosuid nodev noatime noauto_da_alloc,discard,journal_async_commit,errors=panic
+    #wait /dev/block/platform/msm_sdcc.1/by-name/cache
+    #check_fs /dev/block/platform/msm_sdcc.1/by-name/cache ext4
+    #mount ext4 /dev/block/platform/msm_sdcc.1/by-name/cache /cache nosuid nodev noatime noauto_da_alloc,discard,journal_async_commit,errors=panic
 
     #wait /dev/block/platform/msm_sdcc.1/by-name/persist
     #check_fs /dev/block/platform/msm_sdcc.1/by-name/persist ext4
