--- a/META-INF/com/google/android/aroma-config
+++ b/META-INF/com/google/android/aroma-config
@@ -44,7 +44,7 @@
 	"\n\n\n\n\n"+
 	"You Are About To Flash\n"+
 	ini_get("rom_name") + " \nBy\n" + ini_get("rom_author") +
-	"\n\n\n"+
+	"\n\n\n***IMPORTANT: You MUST choose the stock kernel for dual boot to work. If you want to use a custom kernel, you can patch and flash it afterwards.***\n\n\n"+
 	"Version: <#080>" + ini_get("rom_version") + "</#>\n"+
 	"Device: <#080>" + ini_get("rom_device") + "</#>\n\n\n"+
 	"Press Next To Continue The Installation...",
