--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -1,33 +1,44 @@
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+ui_print("Installing ROM as secondary");
+
 #dont kang my work!!!.....again
 ui_print("@ DEVIANT DEVELOPMENT	");
 ui_print("@ Evil Unwizzed S4");
 ui_print(" ");
 ui_print(" ");
-unmount("/system");
 ui_print("	Hacking files on the phone");
 if
     file_getprop("/tmp/aroma/install.prop","selected") == "1"
 then
 set_progress(0.10);
-format("ext4", "EMMC", "/dev/block/mmcblk0p16", "0");
-mount("ext4", "EMMC", "/dev/block/mmcblk0p18", "/cache");
-mount("ext4", "EMMC", "/dev/block/mmcblk0p16", "/system");
-mount("ext4", "EMMC", "/dev/block/mmcblk0p29", "/data");
-ui_print("	Clearing Cache and Dalvik...");
-delete_recursive("/cache");
-delete_recursive("/data/dalvik-cache");
-delete_recursive("/dalvik/dalvik-cache");
 ui_print("	Unleashing the evil");
+
+ui_print("Formatting /system/dual for dual boot");
+mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "/system");
+mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/cache", "/cache");
+delete_recursive("/system/dual");
+delete_recursive("/cache/dual");
+unmount("/system");
+unmount("/cache");
+
+run_program("/tmp/dualboot.sh", "mount-system");
+run_program("/sbin/busybox", "umount", "/data");
+run_program("/tmp/dualboot.sh", "mount-data");
+
+delete_recursive("/data/dalvik-cache");
+
 package_extract_dir("setup", "/tmp");
 set_perm_recursive(0, 0, 0755, 0755, "/tmp");
-run_program("/tmp/fsck.ext4", "-fy", "/dev/block/mmcblk0p16");
-run_program("/tmp/mkfs.ext4", "-O", "^ext_attr,^has_journal,^huge_file", "-L", "SYSTEM", "-b", "4096", "-m", "0", "-F", "/dev/block/mmcblk0p16");
-run_program("/tmp/tune2fs.ext4", "-c", "1", "-m", "0", "-o", "journal_data_writeback", "/dev/block/mmcblk0p16");
-run_program("/tmp/fsck.ext4", "-Dfy", "/dev/block/mmcblk0p16");
-run_program("/tmp/fsck.ext4", "-fy", "/dev/block/mmcblk0p18");
-run_program("/tmp/mkfs.ext4", "-O", "^ext_attr,^has_journal,^huge_file", "-L", "CACHE", "-b", "4096", "-m", "0", "-F", "/dev/block/mmcblk0p18");
-run_program("/tmp/tune2fs.ext4", "-c", "1", "-m", "0", "-o", "journal_data_writeback", "/dev/block/mmcblk0p18");
-run_program("/tmp/fsck.ext4", "-Dfy", "/dev/block/mmcblk0p18");
+# Do not change ext4 default mount options and attributes in dual-boot setup
+#run_program("/tmp/fsck.ext4", "-fy", "/dev/block/mmcblk0p16");
+#run_program("/tmp/mkfs.ext4", "-O", "^ext_attr,^has_journal,^huge_file", "-L", "SYSTEM", "-b", "4096", "-m", "0", "-F", "/dev/block/mmcblk0p16");
+#run_program("/tmp/tune2fs.ext4", "-c", "1", "-m", "0", "-o", "journal_data_writeback", "/dev/block/mmcblk0p16");
+#run_program("/tmp/fsck.ext4", "-Dfy", "/dev/block/mmcblk0p16");
+#run_program("/tmp/fsck.ext4", "-fy", "/dev/block/mmcblk0p18");
+#run_program("/tmp/mkfs.ext4", "-O", "^ext_attr,^has_journal,^huge_file", "-L", "CACHE", "-b", "4096", "-m", "0", "-F", "/dev/block/mmcblk0p18");
+#run_program("/tmp/tune2fs.ext4", "-c", "1", "-m", "0", "-o", "journal_data_writeback", "/dev/block/mmcblk0p18");
+#run_program("/tmp/fsck.ext4", "-Dfy", "/dev/block/mmcblk0p18");
 package_extract_dir("data", "/data");
 package_extract_dir("system", "/system");
 endif;
@@ -36,12 +47,17 @@
     file_getprop("/tmp/aroma/install.prop","selected") == "2"
 then
 set_progress(0.10);
-mount("ext4", "EMMC", "/dev/block/mmcblk0p29", "/data");
-mount("ext4", "EMMC", "/dev/block/mmcblk0p16", "/system");
-ui_print("	Clearing Cache and Dalvik...");
-	delete_recursive("/cache");
-	delete_recursive("/data/dalvik-cache");
-	delete_recursive("/dalvik/dalvik-cache");
+
+mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/cache", "/cache");
+delete_recursive("/cache/dual");
+unmount("/cache");
+
+run_program("/tmp/dualboot.sh", "mount-system");
+run_program("/sbin/busybox", "umount", "/data");
+run_program("/tmp/dualboot.sh", "mount-data");
+
+delete_recursive("/data/dalvik-cache");
+
 endif;
 if
     (file_getprop("/tmp/aroma/install.prop","selected") == "1" || file_getprop("/tmp/aroma/install.prop","selected") == "2")
@@ -260,6 +276,7 @@
 ui_print("	Flashing kernel ");	   
 assert(package_extract_file("boot.img", "/tmp/boot.img"),
        write_raw_image("/tmp/boot.img", "/dev/block/mmcblk0p20"),
+       run_program("/tmp/dualboot.sh", "set-multi-kernel"),
        delete("/tmp/boot.img"));
 endif;
 #Theme
@@ -284,7 +301,6 @@
     if
       file_getprop("/tmp/aroma/theme.prop","selected.2") == "1"
     then
-	  run_program("/sbin/busybox", "mount", "/system");
       ui_print("@ Stock Clock"); 
 	  package_extract_dir("evil/Theme/SClock/vrtheme", "/sdcard/vrtheme");
 	  set_perm(0, 0, 0755, "/sdcard/vrtheme/installtheme.sh");
@@ -294,13 +310,11 @@
       ui_print("	Adding Images");
 	  run_program("/sdcard/vrtheme/installtheme.sh");
 	  run_program("/sdcard/vrtheme/cleanup.sh");
-	  run_program("/sbin/busybox", "umount", "/system"); 
 	  run_program("/sbin/busybox", "umount", "/sdcard"); 
     endif;
     if
       file_getprop("/tmp/aroma/theme.prop","selected.2") == "2"
     then
-	  run_program("/sbin/busybox", "mount", "/system");
       ui_print("@ Center Clock"); 
 	  package_extract_dir("evil/Theme/CClock/vrtheme", "/sdcard/vrtheme");
 	  set_perm(0, 0, 0755, "/sdcard/vrtheme/installtheme.sh");
@@ -310,13 +324,11 @@
       ui_print("	Adding Images");
 	  run_program("/sdcard/vrtheme/installtheme.sh");
 	  run_program("/sdcard/vrtheme/cleanup.sh");
-	  run_program("/sbin/busybox", "umount", "/system"); 
 	  run_program("/sbin/busybox", "umount", "/sdcard"); 
     endif;
     if
       file_getprop("/tmp/aroma/theme.prop","selected.2") == "3"
     then
-	  run_program("/sbin/busybox", "mount", "/system");
       ui_print("@ Offset Clock"); 
 	  package_extract_dir("evil/Theme/OClock/vrtheme", "/sdcard/vrtheme");
 	  set_perm(0, 0, 0755, "/sdcard/vrtheme/installtheme.sh");
@@ -326,13 +338,11 @@
       ui_print("	Adding Images");
 	  run_program("/sdcard/vrtheme/installtheme.sh");
 	  run_program("/sdcard/vrtheme/cleanup.sh");
-	  run_program("/sbin/busybox", "umount", "/system"); 
 	  run_program("/sbin/busybox", "umount", "/sdcard"); 
     endif;
 endif;
-unmount("/system");
-unmount("/cache");
-unmount("/data");
+run_program("/tmp/dualboot.sh", "unmount-system");
+run_program("/tmp/dualboot.sh", "unmount-data");
 ui_print(" ");
 ui_print(" ");
 ui_print(" ");
@@ -350,4 +360,4 @@
 ui_print("@ Donations are appreciated!	");
 ui_print(" ");
 ui_print(" ");
-set_progress(1);
\ No newline at end of file
+set_progress(1);
