From a9f966bfae3d7d11c6681c550d9815fc350771c9 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Sat, 15 Nov 2014 18:14:00 -0500
Subject: [PATCH] boost: Build with NDK 10c

---
 build-android.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build-android.sh b/build-android.sh
index f50c233..319acaa 100755
--- a/build-android.sh
+++ b/build-android.sh
@@ -257,7 +257,7 @@ case "$NDK_RN" in
 		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/${PlatformOS}-x86_64/bin/arm-linux-androideabi-g++
 		TOOLSET=gcc-androidR8e
 		;;
-	"10 (64-bit)"|"10b (64-bit)")
+	"10 (64-bit)"|"10b (64-bit)"|"10c (64-bit)")
 		TOOLCHAIN=${TOOLCHAIN:-arm-linux-androideabi-4.6}
 		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/${PlatformOS}-x86_64/bin/arm-linux-androideabi-g++
 		TOOLSET=gcc-androidR8e
-- 
2.1.3

