--- a/META-INF/com/google/android/update-binary
+++ b/META-INF/com/google/android/update-binary
@@ -663,7 +663,7 @@
 eval "playgames_size=\$pg_${gms}_size"; # Determine size of PlayGames
 
 # Read and save system partition size details
-df=$(busybox df -k /system | tail -n 1);
+df=$(busybox df -k | grep '[ \t]/system$' | tail -n 1);
 case $df in
     /dev/block/*) df=$(echo $df | awk '{ print substr($0, index($0,$2)) }');;
 esac;
