--- configure.orig	2014-12-05 09:24:14.000000000 -0700
+++ configure	2014-12-11 09:24:32.545799224 -0700
@@ -4765,7 +4765,7 @@
     fi
     if [ "$CFG_OPENGL" = "auto" ] || [ "$CFG_OPENGL" = "yes" ]; then
         if compileTest x11/opengl "OpenGL"; then
-            CFG_OPENGL=desktop
+            CFG_OPENGL=angle
         elif compileTest unix/opengles2 "OpenGL ES 2.0"; then
             CFG_OPENGL=es2
         else
@@ -5632,6 +5632,11 @@
     QT_CONFIG="$QT_CONFIG opengles2"
 fi
 
+if [ "$CFG_OPENGL" = "angle" ]; then
+    QCONFIG_FLAGS="$QCONFIG_FLAGS QT_OPENGL_ES QT_OPENGL_ES_2"
+    QT_CONFIG="$QT_CONFIG opengles2 egl angle"
+fi
+
 if [ "$CFG_SHARED" = "yes" ]; then
     QT_CONFIG="$QT_CONFIG shared"
     QTCONFIG_CONFIG="$QTCONFIG_CONFIG shared"
@@ -6588,7 +6593,7 @@
 report_support "  NIS ...................." "$CFG_NIS"
 report_support "  OpenGL / OpenVG:"
 report_support "    EGL .................." "$CFG_EGL"
-report_support "    OpenGL ..............." "$CFG_OPENGL" yes "Desktop OpenGL" es2 "OpenGL ES 2.0+"
+report_support "    OpenGL ..............." "$CFG_OPENGL" yes "Desktop OpenGL" es2 "OpenGL ES 2.0+" angle "ANGLE"
 report_support "    OpenVG ..............." "$CFG_OPENVG-$CFG_OPENVG_SHIVA" yes-yes "ShivaVG" yes-no "native"
 report_support "  PCRE ..................." "$CFG_PCRE" yes "system library" qt "bundled copy"
 if [ -n "$PKG_CONFIG" ]; then
--- src/plugins/platforms/windows/qwindowsglcontext.cpp.orig	2014-12-05 09:24:37.000000000 -0700
+++ src/plugins/platforms/windows/qwindowsglcontext.cpp	2014-12-11 09:27:37.364281924 -0700
@@ -45,7 +45,9 @@
 #include <algorithm>
 
 #include <wingdi.h>
+#ifndef QT_OPENGL_ES_2
 #include <GL/gl.h>
+#endif
 
 // #define DEBUG_GL
 
