--- a/src/tools/bootstrap/bootstrap.pro	2014-01-01 13:22:42.367935800 -0600
+++ b/src/tools/bootstrap/bootstrap.pro	2014-01-02 23:38:20.707687700 -0600
@@ -145,7 +145,7 @@
         ../../corelib/io/qstandardpaths_win.cpp
 }
 
-if(contains(QT_CONFIG, zlib)|cross_compile):include(../../3rdparty/zlib.pri)
+if(contains(QT_CONFIG, zlib)):include(../../3rdparty/zlib.pri)
 else:include(../../3rdparty/zlib_dependency.pri)
 
 win32:LIBS += -luser32 -lole32 -ladvapi32 -lshell32
