--- mkspecs/features/qt_module.prf.orig	2013-07-31 20:37:56.789883201 +0200
+++ mkspecs/features/qt_module.prf	2013-07-31 20:38:06.013034478 +0200
@@ -25,7 +25,7 @@
 
 # Compile as shared/DLL or static according to the option given to configure
 # unless overridden. Host builds are always static
-host_build|staticlib: CONFIG += static
+staticlib: CONFIG += static
 
 host_build {
     QT -= gui   # no host module will ever use gui
--- src/tools/bootstrap-dbus/bootstrap-dbus.pro.orig	2013-12-08 11:09:47.000000000 -0600
+++ src/tools/bootstrap-dbus/bootstrap-dbus.pro	2014-01-05 23:06:13.575543400 -0600
@@ -30,4 +30,4 @@
     ../../dbus/qdbusunixfiledescriptor.cpp
 
 lib.CONFIG = dummy_install
-INSTALLS = lib
+INSTALLS += lib
