From f45790b51450cde0ae74f448a376a9a0bec542f8 Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <chenxiaolong@cxl.epac.to>
Date: Sun, 24 Dec 2017 15:15:01 -0500
Subject: [PATCH] Add Android.mk

---
 Android.mk | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)
 create mode 100644 Android.mk

diff --git a/Android.mk b/Android.mk
new file mode 100644
index 0000000..ad97c5f
--- /dev/null
+++ b/Android.mk
@@ -0,0 +1,37 @@
+LOCAL_PATH:= $(call my-dir)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := libpng
+LOCAL_SRC_FILES := \
+    png.c \
+    pngerror.c \
+    pngget.c \
+    pngmem.c \
+    pngpread.c \
+    pngread.c \
+    pngrio.c \
+    pngrtran.c \
+    pngrutil.c \
+    pngset.c \
+    pngtrans.c \
+    pngwio.c \
+    pngwrite.c \
+    pngwtran.c \
+    pngwutil.c
+LOCAL_CFLAGS += \
+	-std=gnu89 \
+	-Wno-unused-parameter \
+	-ftrapv
+ifneq (,$(filter $(TARGET_ARCH_ABI),armeabi-v7a arm64-v8a))
+	LOCAL_SRC_FILES += \
+		arm/arm_init.c \
+		arm/filter_neon.S \
+		arm/filter_neon_intrinsics.c
+	LOCAL_CFLAGS += -DPNG_ARM_NEON_OPT=2 -fno-integrated-as
+else ifneq (,$(filter $(TARGET_ARCH_ABI),x86 x86_64))
+	LOCAL_SRC_FILES += \
+		contrib/intel/intel_init.c \
+		contrib/intel/filter_sse2_intrinsics.c
+	LOCAL_CFLAGS += -DPNG_INTEL_SSE_OPT=1
+endif
+include $(BUILD_STATIC_LIBRARY)
-- 
2.17.1

